core/src/banking_stage/decision_maker.rs

```diff
@@ -101,17 +94,9 @@ impl DecisionMaker {
             // Node will be leader within ~20 slots, hold the transactions in
             // case it is the only node which produces an accepted slot.
             BufferedPacketsDecision::ForwardAndHold
```

Comment:
I'm not understanding the case where we would even hit the `Hold` case for the old logic.

1. If we didn't know the leader (because we don't have the leader schedule), but in reality we generate the schedule so far in advance relative to how far ahead we look that this seems impossible outside of weird, contrived test cases.
2. If we somehow don't think we will be leader within 20 slots based on ticks but also think we will be leader in 2 slots based on the leader schedule. Is there even a corner case where this could happen?

Reply:
yeah, sorry, #1 makes perfect sense. I was referring to the cases we're removing (#2 and #3)

Reply:
Right, and to summarize my comment - 2 and 3 should NOT happen unless there are other bugs, to the best of my knowledge.